### PR TITLE
Add remote backend

### PR DESCRIPTION
**Pull request by ehabshaaban**
- [x] Add remote backend

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -18,12 +24,12 @@ terraform {
 
     kubernetes = {
       source  = "hashicorp/kubernetes"
-      version = "2.0.1"
+      version = "2.11.0"
```
Comment (omarahm3) [inline versions.tf:27]: Are we sure that this version is stable enough to upgrade to ? if not please do a round of tests just to be 100% sure

File: versions.tf [hunk 2]
```diff
@@ -18,12 +24,12 @@ terraform {
...
     }
 
     helm = {
       source  = "hashicorp/helm"
-      version = "2.0.1"
+      version = "2.5.1"
```
Comment (omarahm3) [inline versions.tf:32]: Same here

File: .github/actions/setup/action.yml [hunk 3]
```diff
@@ -31,4 +31,4 @@ runs:
         
     - name: Terraform init
       shell: bash
-      run: terraform init
\ No newline at end of file
+      run: terraform init
```
Comment (omarahm3) [inline .github/actions/setup/action.yml:34]: What was the actual change here?

File: .github/workflows/terraform-deploy.yml [hunk 4]
```diff
@@ -44,12 +44,12 @@ jobs:
       - name: Terraform init
         uses: ./.github/actions/setup
       
-      - name: Terraform test
-        run: cd test && go test -v -timeout 3000s infra_test.go
-        continue-on-error: true
+      # - name: Terraform test
```
Comment (omarahm3) [inline .github/workflows/terraform-deploy.yml:47]: Don't ever add code comments to production ready code, please remove

File: .github/workflows/terraform-deploy.yml [hunk 5]
```diff
@@ -44,12 +44,12 @@ jobs:
...
 
-      # - name: Terraform deploy
-      #   run: TF_VAR_grafana_admin_password=${{ env.PASSWD }} terraform apply --auto-approve
+      - name: Terraform deploy
+        run: TF_VAR_grafana_admin_password=${{ env.PASSWD }} terraform apply --auto-approve
 
       # - name: Terraform destroy
-      #   run: TF_VAR_grafana_admin_password=${{ env.PASSWD }} terraform destroy --auto-approve
\ No newline at end of file
+      #   run: TF_VAR_grafana_admin_password=${{ env.PASSWD }} terraform destroy --auto-approve
```
Comment (omarahm3) [inline .github/workflows/terraform-deploy.yml:55]: Same here

File: .github/workflows/terraform-deploy.yml [hunk 6]
```diff
@@ -44,12 +44,12 @@ jobs:
...
-        run: cd test && go test -v -timeout 3000s infra_test.go
-        continue-on-error: true
+      # - name: Terraform test
+      #  run: cd test && go test -v -timeout 3000s infra_test.go
+      #  continue-on-error: true
 
-      # - name: Terraform deploy
-      #   run: TF_VAR_grafana_admin_password=${{ env.PASSWD }} terraform apply --auto-approve
+      - name: Terraform deploy
+        run: TF_VAR_grafana_admin_password=${{ env.PASSWD }} terraform apply --auto-approve
```
Comment (omarahm3) [inline .github/workflows/terraform-deploy.yml:52]: why `PASSWD` you're not shortening anything! its just confusing and not even `PASS` as any convention, please rename it to `PASSWORD` instead

**Review by omarahm3 — CHANGES_REQUESTED** [review]

**Comment by omarahm3** [comment]
> * [x]  Add remote backend

WTF is this description? 

Please add proper PR description, so that reviewers can understand actually the reason behind fix/feat/chore/...whatever you're doing.

**Review by mohammedx3 — APPROVED** [review]

**Pull request merged**